src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -71,6 +71,12 @@ public function getEnvPlaceholders()
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
```

Comment:
The array is recursive, but the `+` isn't. So: the 1st `+` adds the new keys to the property, then `+` in the foreach adds the new entries found in previously existing keys.